src/exec_auth.ts

```diff
@@ -99,15 +99,33 @@ export class ExecAuth implements Authenticator {
             exec.env.forEach((elt) => (env[elt.name] = elt.value));
             opts = { ...opts, env };
         }
-        const result = this.execFn(exec.command, exec.args, opts);
-        if (result.error) {
-            throw result.error;
-        }
-        if (result.status === 0) {
-            const obj = JSON.parse(result.stdout.toString('utf8')) as Credential;
-            this.tokenCache[user.name] = obj;
-            return obj;
-        }
-        throw new Error(result.stderr.toString('utf8'));
+
+        return new Promise((accept, reject) => {
+            let stdoutData: string = '';
+            let stderrData: string = '';
+
+            const subprocess = this.execFn(exec.command, exec.args, opts);
+
+            subprocess.stdout.on('data', (data: Buffer | string) => {
+                stdoutData += data.toString('utf8');
+            });
+
+            subprocess.stderr.on('data', (data) => {
+                stderrData += data.toString('utf8');
+            });
+
+            subprocess.on('error', (error) => {
+                throw error;
```

Comment:
Should this call `reject(error)` instead of throwing? It might also be worth putting `error` in a variable instead and then handling the `resolve()`/`reject()` logic in the `'close'` event so that you don't have to handle the case where the promise is both resolved and rejected.

Reply:
thanks, I'll change to delay the reject to the `close` event